app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -164,9 +164,20 @@ fun ActivityDetailsScreen(
 
   val creatorID = activity?.creator ?: ""
   var creator by remember {
-    mutableStateOf(User(creatorID, "anonymous", "anonymous", listOf(), listOf(), "", listOf()))
+    mutableStateOf(User(creatorID, "null", "null", listOf(), listOf(), "", listOf()))
   }
+
   activity?.participants?.firstOrNull { it.id == creatorID }?.let { creator = it }
+  if (creator.name == "null") {
+    profileViewModel.getUserData(
+        creatorID,
+        onResult = {
+          if (it != null) {
+            creator = it
+          }
+        })
+  }
+  Log.d("ActivityDetailsScreen", "ActivityDetails Creator: $creator")
```

Comment:
This was the missing logic, before this, it was displaying "Anonymous" when a creator is not enrolled in his own activity.